boards: nxp: add support for FRDM-IMX93

CkovMk opened the pull request:
Support FRDM-IMX93

nordicjm commented on boards/nxp/frdm_imx93/Kconfig:
```diff
@@ -0,0 +1,4 @@
+# FRDM-IMX93 board configuration
+
+# Copyright 2025 NXP
```
file to go

nordicjm commented on boards/nxp/frdm_imx93/frdm_imx93-pinctrl.dtsi:
```diff
@@ -0,0 +1,132 @@
+			slew-rate = "fast";
+			drive-strength = "x6";
+		};
+
```
```suggestion
```

nordicjm commented on boards/nxp/frdm_imx93/frdm_imx93-pinctrl.dtsi:
```diff
@@ -0,0 +1,132 @@
+		};
+
+	};
+
```
```suggestion
```
fix in whole pr

nordicjm commented on boards/nxp/frdm_imx93/frdm_imx93_mimx9352_a55.dts:
```diff
@@ -0,0 +1,150 @@
+
+	leds {
+		compatible = "gpio-leds";
+		led_r: led_r {
```
line 43, 47, 51, missing newlines, fix in whole pr

kartben commented on boards/nxp/frdm_imx93/doc/index.rst:
```diff
@@ -0,0 +1,164 @@
+References
+==========
+
+More information can refer to NXP official website:
```
this sentence is not grammatically correct.  Also, I think you guys typically include the following -- cc @DerekSnell 

```
.. include:: ../../common/board-footer.rst
   :start-after: nxp-board-footer
```

kartben reviewed (changes requested):
Also please add a board image -- see attached for convenience
[frdm_imx93.webp.zip](https://github.com/user-attachments/files/21342575/frdm_imx93.webp.zip)

nordicjm commented on boards/nxp/frdm_imx93/frdm_imx93_mimx9352_a55.dts:
```diff
@@ -0,0 +1,153 @@
+&lpuart1 {
+	status = "disabled";
+	current-speed = <115200>;
+	/* clocks = <&ccm IMX_CCM_UART4_CLK 0x6c 24>; */
```
is this needed?

nordicjm commented on boards/nxp/frdm_imx93/frdm_imx93_mimx9352_a55.dts:
```diff
@@ -0,0 +1,153 @@
+&lpuart2 {
+	status = "okay";
+	current-speed = <115200>;
+	/* clocks = <&ccm IMX_CCM_UART4_CLK 0x6c 24>; */
```
etc. ?

LaurentiuM1234 commented on boards/nxp/frdm_imx93/frdm_imx93_mimx9352_a55.dts:
```diff
@@ -0,0 +1,151 @@
+	};
+};
+
+&lpuart1 {
```
I'd suggest ordering these by node label name: https://docs.kernel.org/devicetree/bindings/dts-coding-style.html

also, why add this node? you're keeping it disabled?

CkovMk replied:
This one can be removed. I just copied the files from i.MX93 EVK and account for the differences.

LaurentiuM1234 commented on boards/nxp/frdm_imx93/frdm_imx93_mimx9352_a55.dts:
```diff
@@ -0,0 +1,151 @@
+	pinctrl-names = "default";
+};
+
+&gpio1{
```
space between the node label name and bracket pls. Also, isn't this already enabled in the DTSI?

CkovMk replied:
GPIOs are not enabled by default in dtsi - peripherals are all disabled. I'd like to keep this consistent.

LaurentiuM1234 commented on boards/nxp/frdm_imx93/frdm_imx93_mimx9352_a55.dts:
```diff
@@ -0,0 +1,151 @@
+	status = "okay";
+};
+
+&gpio2{
```
same

LaurentiuM1234 commented on boards/nxp/frdm_imx93/frdm_imx93_mimx9352_a55.dts:
```diff
@@ -0,0 +1,151 @@
+	status = "okay";
+};
+
+&gpio3{
```
same

LaurentiuM1234 commented on boards/nxp/frdm_imx93/frdm_imx93_mimx9352_a55.dts:
```diff
@@ -0,0 +1,151 @@
+	status = "okay";
+};
+
+&gpio4{
```
same

LaurentiuM1234 commented on boards/nxp/frdm_imx93/frdm_imx93_mimx9352_a55.dts:
```diff
@@ -0,0 +1,151 @@
+		reg = <0xd0000000 DT_SIZE_M(1)>;
+	};
+
+	aliases {
```
maybe place this before "chosen" for visibility?

LaurentiuM1234 commented on boards/nxp/frdm_imx93/frdm_imx93_mimx9352_a55.dts:
```diff
@@ -0,0 +1,151 @@
+
+/ {
+	model = "NXP i.MX93 A55";
+	compatible = "fsl,mimx93";
```
maybe it would be a good idea to try and keep these consistent with Linux? i.e.: `compatible = "fsl,imx93-11x11-frdm", "fsl,imx93";`

LaurentiuM1234 commented on boards/nxp/frdm_imx93/frdm_imx93_mimx9352_a55.dts:
```diff
@@ -0,0 +1,151 @@
+
+#include <nxp/nxp_mimx93_a55.dtsi>
+#include "frdm_imx93-pinctrl.dtsi"
+#include <zephyr/dt-bindings/input/input-event-codes.h>
```
keep alphabetical order please with "frdm_imx93-pinctrl.dtsi" at the end

CkovMk replied:
I think it's better to group the includes as "dtsi" and "headers". If we have multiple headers we can keep alphabetical order within its category.

CkovMk replied:
won't fix, as explained above

kartben replied:
yes, fully agreed that it's just fine as-is, and how it's done in most boards (including NXP :) )

LaurentiuM1234 commented on boards/nxp/frdm_imx93/frdm_imx93-pinctrl.dtsi:
```diff
@@ -0,0 +1,130 @@
+	uart1_default: uart1_default {
+		group0 {
+			pinmux = <&iomuxc1_uart1_rxd_lpuart_rx_lpuart1_rx>,
+				<&iomuxc1_uart1_txd_lpuart_tx_lpuart1_tx>;
```
keep the second phandle aligned with the first one. This comment applies to the whole DTSI.

LaurentiuM1234 commented on boards/nxp/frdm_imx93/frdm_imx93_mimx9352_a55.dts:
```diff
@@ -0,0 +1,151 @@
+};
+
+&lpi2c1 {
+	status = "disabled";
```
why add this node? you're keeping it disabled?

CkovMk replied:
Remove it for now. There used to be a GPIO expander here which driver is not present upstream.

LaurentiuM1234 commented on boards/nxp/frdm_imx93/frdm_imx93_mimx9352_a55.dts:
```diff
@@ -0,0 +1,151 @@
+};
+
+&lpuart2 {
+	status = "okay";
```
please keep status as last property: https://docs.kernel.org/devicetree/bindings/dts-coding-style.html. This comment applies to all nodes which don't abide by this.

LaurentiuM1234 commented on boards/nxp/frdm_imx93/frdm_imx93_mimx9352_a55.dts:
```diff
@@ -0,0 +1,151 @@
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+
+/ {
+	model = "NXP i.MX93 A55";
```
how about  `model = "NXP i.MX93 11X11 FRDM board A55 core"`? basically, the model name used in Linux + the differentiation between A55 and the M core

CkovMk replied:
rename to "NXP FRDM-IMX93 A55".

LaurentiuM1234 commented on boards/nxp/frdm_imx93/frdm_imx93_mimx9352_a55.dts:
```diff
@@ -0,0 +1,151 @@
+	chosen {
+		zephyr,console = &lpuart2;
+		zephyr,shell-uart = &lpuart2;
+		/* sram node actually locates at DDR DRAM */
```
do we still need this comment? we've been using the DDR for a while now...

LaurentiuM1234 replied:
please drop. This comment adds no new information. It is already evident that we're not using SRAM from the node name and its address space.

kartben commented on boards/nxp/frdm_imx93/doc/index.rst:
```diff
@@ -0,0 +1,164 @@
+--------------------------
+
+The user buttons USER_BTN1 and USER_BTN2 is connected to onboard GPIO expander
+PCAL6524 by default, but can be changed to connect to FRDM-IMX93 GPIO. An devicetree
```
```suggestion
PCAL6524 by default, but can be changed to connect to FRDM-IMX93 GPIO. A devicetree
```

LaurentiuM1234 reviewed (commented):
Some more comments below. Also, please let people mark their own comments as resolved.

CkovMk committed:
dts: nxp_mimx93_a55: make gpio disabled by default

GPIO should be enabled in boards. Disable them by default.

Signed-off-by: Chekhov Ma <chekhov.ma@nxp.com>

whbo158 committed:
boards: nxp: add support for platform FRDM-IMX93

Add initial support for FRDM-IMX93 board

Building hello_world:
west build -p always -b frdm_imx93/mimx9352/a55 samples/hello_world

Signed-off-by: Hongbo Wang <hongbo.wang@nxp.com>
Signed-off-by: Chekhov Ma <chekhov.ma@nxp.com>
